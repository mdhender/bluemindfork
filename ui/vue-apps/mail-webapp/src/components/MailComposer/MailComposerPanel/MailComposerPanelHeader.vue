<template>
    <div class="mail-composer-panel-header card-header d-flex justify-content-between align-items-center rounded-top">
        <span v-bm-tooltip.hover.ds500 class="text-nowrap text-truncate" :title="title">{{ title }}</span>
        <bm-button-close v-if="closeable" text-variant="light" @click="$emit('remove')" />
    </div>
</template>

<script>
import { BmButtonClose, BmTooltip } from "@bluemind/styleguide";

export default {
    name: "MailComposerPanelHeader",
    components: { BmButtonClose },
    directives: { BmTooltip },
    props: {
        title: {
            type: String,
            required: false,
            default: ""
        },
        closeable: {
            type: Boolean,
            required: false,
            default: true
        }
    }
};
</script>
<style lang="scss">
@import "~@bluemind/styleguide/css/_variables";

.mail-composer-panel-header .close {
    // bootstrap does not adapt the shadow's color
    text-shadow: 0 1px 0 $black;

    // compensates the too big letter-spacing (dont touch letter-spacing since it modifies the click box)
    margin-right: -0.2em;
}
</style>
