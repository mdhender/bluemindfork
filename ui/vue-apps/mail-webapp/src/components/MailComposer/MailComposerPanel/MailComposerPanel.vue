<template>
    <div class="d-flex flex-column bg-white mail-composer-panel">
        <slot name="header">
            <mail-composer-panel-header :title="title" :closeable="closeable" @remove="$emit('remove')" />
        </slot>
        <div class="flex-grow-1 d-flex flex-column">
            <slot name="body" />
        </div>
        <div class="p-2 border-top mail-composer-panel-footer">
            <slot name="footer" />
        </div>
    </div>
</template>

<script>
import MailComposerPanelHeader from "./MailComposerPanelHeader";

export default {
    name: "MailComposerPanel",
    components: {
        MailComposerPanelHeader
    },
    props: {
        title: {
            type: String,
            required: false,
            default: ""
        },
        closeable: {
            type: Boolean,
            required: false,
            default: true
        }
    }
};
</script>
