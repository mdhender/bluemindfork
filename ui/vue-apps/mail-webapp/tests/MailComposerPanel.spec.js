import { mount } from "@vue/test-utils";
jest.mock("@bluemind/styleguide/css/_variables.scss", () => ({ iconsColors: "" }));
import MailComposerPanel from "../src/components/MailComposer/MailComposerPanel";

describe("MailComposerPanelHeader", () => {
    let closeButtonSelector = ".close";
    let title = "My Panel Title";
    let bodyContent = "<span>My body is a temple.</span>";
    let footerContent = "<button>Submit</button>";

    function defaultMount() {
        return mount(MailComposerPanel, {
            propsData: {
                title: title
            },
            slots: {
                body: bodyContent,
                footer: footerContent
            },
            sync: false
        });
    }

    let regexCloseButton = new RegExp(/<button type="button".*class="close/);
    let regexBodyContent = new RegExp(/<span>\s*My body is a temple.\s*<\/span>/);
    let regexFooterContent = new RegExp(/<button>\s*Submit\s*<\/button>/);

    test("is a Vue instance", () => {
        expect(defaultMount().vm).toBeTruthy();
    });

    test("MailComposerPanel should match snapshot", () => {
        expect(defaultMount().vm.$el).toMatchSnapshot();
    });

    test("Simple MailComposerPanel display: a title, a body and a footer", () => {
        const wrapper = defaultMount();
        expect(wrapper.text()).toContain(title);
        expect(wrapper.html()).toMatch(regexCloseButton);
        expect(wrapper.html()).toMatch(regexBodyContent);
        expect(wrapper.html()).toMatch(regexFooterContent);
    });

    test("When clicking on close button, a remove event is fired", () => {
        const wrapper = defaultMount();
        wrapper.find(closeButtonSelector).trigger("click");
        expect(wrapper.emitted().remove).toBeTruthy();
    });

    test("No close button", () => {
        const wrapper = mount(MailComposerPanel, {
            propsData: {
                title: title,
                closeable: false
            }
        });
        expect(wrapper.text()).toContain(title);
        expect(wrapper.html()).not.toMatch(regexCloseButton);
    });
});
