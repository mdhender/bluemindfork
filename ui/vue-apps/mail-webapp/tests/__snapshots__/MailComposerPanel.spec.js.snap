// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MailComposerPanelHeader MailComposerPanel should match snapshot 1`] = `
<div
  class="d-flex flex-column bg-white mail-composer-panel"
>
  <div
    class="mail-composer-panel-header card-header d-flex justify-content-between align-items-center rounded-top"
  >
    <span
      class="text-nowrap text-truncate"
      title="My Panel Title"
    >
      My Panel Title
    </span>
     
    <button
      aria-label="Close"
      class="close text-light"
      type="button"
    >
      Ã—
    </button>
  </div>
   
  <div
    class="flex-grow-1 d-flex flex-column"
  >
    <span>
      My body is a temple.
    </span>
  </div>
   
  <div
    class="p-2 border-top mail-composer-panel-footer"
  >
    <button>
      Submit
    </button>
  </div>
</div>
`;
